internal/tiltfile/k8s.go

```diff
@@ -324,32 +324,39 @@ func podLabelsFromStarlarkValue(v starlark.Value) ([]labels.Selector, error) {
 	}
 }
 
-func (s *tiltfileState) k8sKind(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var kind string
-	if err := starlark.UnpackArgs(fn.Name(), args, nil,
-		"kind", &kind,
+func (s *tiltfileState) k8sImageJsonPath(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var kind, name, namespace string
+	var imageJSONPath starlark.Value
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs,
+		"path", &imageJSONPath,
+		"kind?", &kind,
+		"name?", &name,
+		"namespace?", &namespace,
 	); err != nil {
 		return nil, err
 	}
 
-	// unpacked separately so that this is a keyword-only arg
-	var imageJSONPath starlark.Value
-	if err := starlark.UnpackArgs(fn.Name(), nil, kwargs,
-		"image_json_path", &imageJSONPath,
-	); err != nil {
-		return nil, err
+	if kind == "" && name == "" && namespace == "" {
+		return nil, errors.New("at least one of kind, name, or namespace must be specified")
 	}
 
 	values := starlarkValueOrSequenceToSlice(imageJSONPath)
 	var paths []string
 	for _, v := range values {
 		s, ok := v.(starlark.String)
 		if !ok {
-			return nil, fmt.Errorf("image_json_path must be a string or list of strings, found a list containing value '%+v' of type '%T'", v, v)
+			return nil, fmt.Errorf("path must be a string or list of strings, found a list containing value '%+v' of type '%T'", v, v)
 		}
 		paths = append(paths, s.String())
 	}
-	s.k8sImageJsonPathsByKind[kind] = paths
+
+	k := imageJSONPathSelector{
+		kind:      kind,
+		name:      name,
+		namespace: namespace,
+	}
+
+	s.k8sImageJSONPaths[k] = paths
```

Comment:
not a huge deal, but is it maybe worth it to try and parse the image path here so if it's invalid, we throw an error at the line the user specified it and not from some random place during execution?

Reply:
yes, good idea!